src/config.ts

```diff
@@ -45,6 +46,15 @@ export class CacheConfig {
    */
   static async new(): Promise<CacheConfig> {
     const self = new CacheConfig();
+    const source = core.getState(STATE_CONFIG);
+    if (source !== "") {
+      // Post action, use what we calculated in the main action ensuring consistency.
+      Object.assign(self, JSON.parse(source));
+      self.workspaces = self.workspaces
+        .map((w: any) => new Workspace(w.root, w.target));
```

Comment:
Instead of this dance, I would rather make an explicit ser/deserialization step.

Reply:
I think I known what you're looking for, will update.